tests/AbstractFactoryDecoratorTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace byrokrat\id;
+
+use PHPUnit\Framework\TestCase;
+
+class AbstractFactoryDecoratorTest extends TestCase
+{
+    public function testConstruct()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {}
+        };
+
+        $this->assertInstanceOf(IdFactory::class, $factory);
```

Comment:
This should be `IdFactoryInterface`. The fact that `AbstractFactoryDecorator` subclass `IdFactory` is a blast from the past That should be `implements IdFactoryInterface` in 2.0.